x-pack/filebeat/input/entityanalytics/provider/azuread/fetcher/group.go

```diff
@@ -11,9 +11,11 @@ type MemberType int
 
 const (
 	// MemberUser is a user.
-	MemberUser MemberType = iota
+	MemberUser MemberType = iota + 1
```

Comment:
Why `+1` ?

Reply:
This prevents accidental zero values from being valid.

Reply:
Should the comment reflect the info?

Reply:
I doesn't add anything for the reader it's just standard defensive programming.

Reply:
Yeah...  Just to let other programmers know why this addition happened :-) unless there is a common aura among the programmers :-)

Reply:
They have blame and it will link to this discussion.

Reply:
> unless there is a common aura among the programmers :-)

It's a common pattern. My rough summary...

- Use `iota` when you want the zero value to the default.
- Use `iota + 1` when you don't want the zero value to be a valid enum value.
- And [sometimes](https://github.com/search?q=path%3A*.go+%2Finvalid%5Cw%2B+%5Cw%2B+%3D+iota%2F&type=code) you will see an explicit invalid zero value like `invalidFooType FooType = iota`.